internal/core/metadata/v2/application/device.go

```diff
@@ -0,0 +1,51 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package application
+
+import (
+	"context"
+	"fmt"
+
+	v2MetadataContainer "github.com/edgexfoundry/edgex-go/internal/core/metadata/v2/bootstrap/container"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/correlation"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
+	"github.com/edgexfoundry/go-mod-core-contracts/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
+
+// The AddDevice function accepts the new device model from the controller function
+// and then invokes AddDevice function of infrastructure layer to add new device
+func AddDevice(d models.Device, ctx context.Context, dic *di.Container) (id string, edgeXerr errors.EdgeX) {
+	dbClient := v2MetadataContainer.DBClientFrom(dic.Get)
+	lc := container.LoggingClientFrom(dic.Get)
+
+	exists, edgeXerr := dbClient.DeviceServiceExistByName(d.ServiceName)
+	if edgeXerr != nil {
+		return id, errors.NewCommonEdgeXWrapper(edgeXerr)
+	} else if !exists {
+		return id, errors.NewCommonEdgeX(errors.KindEntityDoesNotExist, fmt.Sprintf("device service '%s' not exists", d.ServiceName), nil)
```

Comment:
Change `device service '%s' not exists` to `device service '%s' does not exists`

Reply:
Modified.